packages/auth/src/AuthProvider.tsx

```diff
@@ -187,18 +186,26 @@ export class AuthProvider extends React.Component<
    * If the user is assigned any of the provided list of roles,
    * the hasRole is considered to be true.
    */
-  hasRole = (role: string | string[]): boolean => {
+  hasRole = (roleToCheck: string | string[]): boolean => {
     if (
-      typeof role !== 'undefined' &&
+      typeof roleToCheck !== 'undefined' &&
       this.state.currentUser &&
       this.state.currentUser.roles
     ) {
-      if (typeof role === 'string') {
-        return this.state.currentUser.roles?.includes(role) || false
+      if (typeof roleToCheck === 'string') {
+        return this.state.currentUser.roles?.includes(roleToCheck) || false
```

Comment:
Don't we need an `isArray` check on `currentUser.roles` here as well? As you added below

Reply:
Ummm in theory, if you passed a function or something, but we assume that you will pass either a string or an array, and don't do exhaustive checks for that. Think thats ok!

Reply:
What I'm saying is if `roleToCheck` is a string, and `this.state.currentUser.roles` is a string we should not do `includes(roleToCheck)` but rather use `===`.

Reply:
Unless there's something I don't understand about the code here....

Reply:
That's true, if you had `admin_user` and `user`, it'd return `true` in both cases if you checked for the role `user` with `includes()` 😬

Reply:
So with the current setup there's 4 cases it needs to handle:

1. currentUser.roles is a string, roleToCheck is a string
2. currentUser.roles is an array, roleToCheck is a string
3. currentUser.roles is a string, roleToCheck is an array
4. currentUser.roles is an array, roleToCheck is an array

I feel like we're only covering 3 of these in here aren't we?

Reply:
That check on lines 195-196 would cause the partial string problem if `currentUser.roles` is itself a string, but if it changes to `===` then it'll fail if currentUser.roles. is an array.

Reply:
Exactly, we have 4 cases it needs to handle, but only three branches in the code. I'd add another one to (explicitly) handle the missing case too

Reply:
@cannikin, @Tobbe how about this one?